sklearn/utils/stats.py

```diff
@@ -7,30 +7,35 @@
 
 
 def _weighted_percentile(array, sample_weight, percentile=50):
-    """Compute weighted percentile
+    """Compute the lower value(s) at a given weighted percentile.
 
-    Computes lower weighted percentile. If `array` is a 2D array, the
-    `percentile` is computed along the axis 0.
+    If `array` is a 2D array, the `values` are selected along axis 0.
 
         .. versionchanged:: 0.24
             Accepts 2D `array`.
 
+        .. versionchanged:: 1.6
+            Supports handling of `NaN` values. For `NaN` inputs, their corresponding
+            weights are set to 0 in `sample_weight`.
+            Percentiles that point to `NaN` values are redirected to the next lower
+            value if it exists.
+
     Parameters
     ----------
     array : 1D or 2D array
         Values to take the weighted percentile of.
 
     sample_weight: 1D or 2D array
-        Weights for each value in `array`. Must be same shape as `array` or
-        of shape `(array.shape[0],)`.
+        Weights for each value in `array`. Must be same shape as `array` or of shape
+        `(array.shape[0],)`.
 
     percentile: int or float, default=50
-        Percentile to compute. Must be value between 0 and 100.
+        Percentile to compute. Must be between 0 and 100.
```

Comment:
```suggestion
        The level (or rank) of the percentile to compute. Must be between 0 and 100.
```

Reply:
fyi: wikipedia is not a good reference for quantiles.